ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -564,21 +564,9 @@ FEDataManager::reinitElementMappings()
 
     // Reset the mappings between grid patches and active mesh
     // elements.
-    //
-    // TODO: for now we simply set all patches on all levels except the finest
-    // to have no elements.
-    for (int ln = 0; ln <= d_max_level_number; ++ln)
-    {
-        Pointer<PatchLevel<NDIM> > level = d_hierarchy->getPatchLevel(ln);
-        d_active_patch_elem_map[ln].resize(level->getProcessorMapping().getNumberOfLocalIndices());
-        d_active_patch_node_map[ln].resize(level->getProcessorMapping().getNumberOfLocalIndices());
-    }
-
-    // TODO: collectActivePatchElements still assumes every element is on the
-    // same patch level.
-    collectActivePatchElements(d_active_patch_elem_map.back(), d_max_level_number);
     for (int ln = 0; ln <= d_max_level_number; ++ln)
     {
+        collectActivePatchElements(d_active_patch_elem_map[ln], ln, ln, ln);
```

Comment:
This is the main change in the patch - we now actually assign elements to their requested levels.